File: project-euler/problem-3/solution-3.js
function getPrimeFactors(number) {
  let current = number;
  const factors = [];

  while (current !== 1) {
    for (let i = 2; ; i++) {
      if (current % i === 0) {
        current = current / i;
        factors.push(i);
        break;
      }
    }
  }

  return factors;
}

console.log(Math.max(...getPrimeFactors(600851475143))); // 6857

File: project-euler/problem-3/solution-3.c

#include <stdio.h>

int main()
{
    long num = 600851475143;
    int max = 2;
    int i;
    while (num != 1)
    {
        for (i = 2;; i++)
        {
            if (num % i == 0)
            {
                if (i > max)
                {
                    max = i;
                }
                num = num / i;
                break;
            }
        }
    }

    printf("%d", max); // 6857
    return 0;
}
